spec.md

```diff
@@ -537,35 +537,33 @@ message VolumeInfo {
   // NFS share). If set, it MUST be non-zero.
   uint64 capacity_bytes = 1;
 
-  // Contains identity information for the created volume. This field is  
+  // Contains identity information for the created volume. This field is
   // REQUIRED. The identity information will be used by the CO in
   // subsequent calls to refer to the provisioned volume.
-  VolumeID id = 4;
-
-  // Metadata of the created volume. This field is OPTIONAL. If set, the
-  // CO SHALL pass this information along with the `id` to subsequent
-  // calls.
-  VolumeMetadata metadata = 5;
+  VolumeHandle handle = 2;
 }
 
-// The identity of the volume.
-message VolumeID {
-  // The identity of the provisioned volume specified by the Plugin in
-  // the form of key-value pairs. This field is REQUIRED. Given this
-  // information will be passed around by the CO, it is RECOMMENDED that
-  // each Plugin keeps this information as small as possible.
-  map<string, string> values = 1;
-}
-
-// The metadata information about the volume.
-message VolumeMetadata {
-  // The metadata information about the provisioned volume specified by
-  // the Plugin in the form of key-value pairs. This field is OPTIONAL.
-  // This field MAY contain sensitive information. Therefore, the CO
-  // MUST NOT leak this information to untrusted entities. Given this
-  // information will be passed around by the CO, it is RECOMMENDED that
-  // each Plugin keeps this information as small as possible.
-  map<string, string> values = 1;
+// VolumeHandle objects are generated by Plugin implementations to
+// serve two purposes: first, to efficiently identify a volume known
+// to the Plugin implementation; second, to capture additional, opaque,
+// possibly sensitive information associated with a volume. It SHALL
+// NOT change over the lifetime of the volume and MAY be safely cached
+// by the CO.
+// Since this object will be passed around by the CO, it is RECOMMENDED
+// that each Plugin keeps the information herein as small as possible.
+message VolumeHandle {
+
+  // ID is the identity of the provisioned volume specified by the
+  // Plugin. This field is REQUIRED. 
+  // This information SHALL NOT be considered sensitive such that, for
+  // example, the CO MAY generate log messages that include this data.
+  string id = 1;
+
+  // Metadata captures additional, possibly sensitive information about
```

Comment:
nit: `additional, possibly sensitive, information`

Reply:
I am not sure i agree with this nit.  To me it reads better without the extra comma.

Reply:
Having a single comma suggests that "Metadata captures additional" is an introductory clause. But, "possibly sensitive information about..." can't stand on it's on (it is incomplete without the first part of the sentence).

Instead, "possibly sensitive" is nonessential information. And the sentence reads find without it.

From https://owl.english.purdue.edu/owl/owlprint/607/:

```
Commas with Nonessential Elements

Some modifying elements of a sentence are essential, restricting the meaning of a modified term, while others are nonessential and don't restrict the modified term's meaning. These nonessential elements, which can be words, phrases, or clauses, are set off with commas.

Rule: Use commas before and after nonessential words, phrases, and clauses, that is, elements embedded in the sentence that interrupt it without changing the essential meaning.
```

So I would offset "possibly sensitive" with two commas.

I'm being very pedantic here, so feel free to disregard =)

Reply:
ack